File: app/src/main/java/com/example/moneyspendcalculator/ObserverPattern/Observerable.java
package com.example.moneyspendcalculator.ObserverPattern;

import java.util.ArrayList;

public class Observerable {
    private ArrayList<Observer> channels = new ArrayList<>();

    public void addObserver(Observer channel) {
        this.channels.add(channel);
    }

    public void removeObserver(Observer channel) {
        this.channels.remove(channel);
    }

    public void Invoke() {
        for (Observer channel : this.channels) {
            channel.update();
        }
    }
}

File: app/src/main/java/com/example/moneyspendcalculator/ObserverPattern/Observerable.kt
package com.example.moneyspendcalculator.ObserverPattern

class Observerable {
    private val channels = ArrayList<Observer>()
    fun addObserver(channel: Observer) {
        channels.add(channel)
    }

    fun removeObserver(channel: Observer) {
        channels.remove(channel)
    }

    fun Invoke() {
        for (channel in channels) {
            channel.update()
        }
    }
}